internal/test/framework/framework.go

```diff
@@ -215,22 +202,29 @@ func (f *TestFramework) getOpenShiftOperatorClient(config *restclient.Config) er
 // are running against. For example: the kubernetes API server endpoint https://api.abc.devcluster.openshift.com:6443
 // gets converted to abc.devcluster.openshift.com
 func (f *TestFramework) getClusterAddress(config *restclient.Config) error {
-	if config.Host == "" {
-		return fmt.Errorf("API server has empty host name")
+	host, err := f.OSConfigClient.ConfigV1().Infrastructures().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to get cluster infrastructure resource: %v", err)
+	}
+	// get API server internal url of format https://api-int.abc.devcluster.openshift.com:6443
+	if host.Status.APIServerInternalURL == "" {
+		return fmt.Errorf("could not get host name for the kubernetes api server")
 	}
 
-	clusterEndPoint, err := url.Parse(config.Host)
+	kubeAPIServerEndpoint := host.Status.APIServerInternalURL
```

Comment:
we are not using this var anywhere else can directly use the value in clusterEndPoint, err := url.Parse(host.Status.APIServerInternalURL)